### PR TITLE
Kev/3_popup

### PR DESCRIPTION
**Pull request by junhaow1**
# Pull Request Details

## What issue does this PR address

- [A one line description of this PR]

- Link to associated issue: #3 

## Checklist

[Use the check-list below to ensure your branch is ready for PR]

- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint errors (`make prep` or `flutter analyze lib`
                       python: use pre-commit ruff hook to check on commit)
- [ ] No lint check errors related to your changes
- [ ] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [x] Chrome
  - [ ] iOS
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [x] Merge dev into the branch
- [x] Resolve any conflicts
- [x] Add one line summary into CHANGELOG.md
- [x] Bump appropriate version number
- [ ] Merge PR into dev

### CODE REVIEW

**Comment by cdawei** [comment]
Hi Kevin, I got this error when trying to build `keypod` using your branches (`kev/3_popup_login` for `keypod`, and `kev/3_popup` for `solidpod`
```bash
CMake Error at /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:463 (message):
  A required package was not found
Call Stack (most recent call first):
  /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:643 (_pkg_check_modules_internal)
  flutter/ephemeral/.plugin_symlinks/flutter_secure_storage/linux/CMakeLists.txt:14 (pkg_check_modules)
```
Any ideas?

PS: I could successfully run the `keypod` app using the `dev` branch (together with the `dev` branch of `solidpod`). My flutter version is `3.19.2`.

**Comment by junhaow1** [comment]
> Hi Kevin, I got this error when trying to build `keypod` using your branches (`kev/3_popup_login` for `keypod`, and `kev/3_popup` for `solidpod`
> 
> ```shell
> CMake Error at /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:463 (message):
>   A required package was not found
> Call Stack (most recent call first):
>   /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:643 (_pkg_check_modules_internal)
>   flutter/ephemeral/.plugin_symlinks/flutter_secure_storage/linux/CMakeLists.txt:14 (pkg_check_modules)
> ```
> 
> Any ideas?
> 
> PS: I could successfully run the `keypod` app using the `dev` branch (together with the `dev` branch of `solidpod`). My flutter version is `3.19.2`.
let me have a look :)

**Comment by junhaow1** [comment]
> Hi Kevin, I got this error when trying to build `keypod` using your branches (`kev/3_popup_login` for `keypod`, and `kev/3_popup` for `solidpod`
> 
> ```shell
> CMake Error at /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:463 (message):
>   A required package was not found
> Call Stack (most recent call first):
>   /snap/flutter/145/usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:643 (_pkg_check_modules_internal)
>   flutter/ephemeral/.plugin_symlinks/flutter_secure_storage/linux/CMakeLists.txt:14 (pkg_check_modules)
> ```
> 
> Any ideas?
> 
> PS: I could successfully run the `keypod` app using the `dev` branch (together with the `dev` branch of `solidpod`). My flutter version is `3.19.2`.

Flutter 3.19.0 • channel stable is the my version, it was working fine on my machine. Cannot replicate the error you meet. I'll change some codes first as Graham mentioned 1 hour ago that the authdata should be saved locally in solidpod. I'll let you know tomorrow when it is done so that you can review :)

**Comment by gjwgit** [comment]
- [ ] utilise InitialisPod() if required
- [ ] save auth data/token for use on other pages as in SolidLogin()

**Comment by gjwgit** [comment]
Not currently ready for review.

**Comment by junhaow1** [comment]
Hi @anushkavidanage This PR is ready to be merged. The checkboxes are done. :)

**Review by anushkavidanage — APPROVED** [review]

File: pubspec.yaml [hunk 1]
```diff
@@ -39,8 +41,12 @@ dev_dependencies:
   # Not yet use solid-lints. It requires analyze 6.2.0 and
   # dart-code-metrics fails.
   # solid_lints: ^0.1.1
+  riverpod_generator: ^2.4.0
```
Comment (cdawei) [inline pubspec.yaml:44]: Do we use `riverpod_generator` anywhere?

Comment (junhaow1) [reply pubspec.yaml:44]: removing now

**Review by cdawei — APPROVED** [review]
Hi Kevin, the change looks good to me. I didn't see where you would retrieve the serialised `authData` from the secure storage (or is it not necessary for this PR?). Please clarify (and also the `riverpod_generator` dependency).

**Review by gjwgit — COMMENTED** [review]
Was a `make prep` done? When I do it there are some auto fixes and a few complaints. Unused code and files?

**Comment by junhaow1** [comment]
> Hi Kevin, the change looks good to me. I didn't see where you would retrieve the serialised `authData` from the secure storage (or is it not necessary for this PR?). Please clarify (and also the `riverpod_generator` dependency).
Retrieving the serialised authData from the secure storage is what I did in Bstim project(not needed for this PR), riverpod is unnecessary, I've removed it.

**Comment by junhaow1** [comment]
> Was a `make prep` done? When I do it there are some auto fixes and a few complaints. Unused code and files?

doing now

**Comment by cdawei** [comment]
> > Hi Kevin, the change looks good to me. I didn't see where you would retrieve the serialised `authData` from the secure storage (or is it not necessary for this PR?). Please clarify (and also the `riverpod_generator` dependency).
> > Retrieving the serialised authData from the secure storage is what I did in Bstim project(not needed for this PR), riverpod is unnecessary, I've removed it.

I guess the logic for the popup login would be something like:
1. try to retrieve the `authData` from secure storage
2. if step 1 is successful, do nothing
3. otherwise, show the popup login window and do the normal authentication

Is this the agreed logic @junhaow1 and @gjwgit ?

We may need to check token expiration and refresh the access token if expired. But this could be for different issue.

**Comment by junhaow1** [comment]
> > > Hi Kevin, the change looks good to me. I didn't see where you would retrieve the serialised `authData` from the secure storage (or is it not necessary for this PR?). Please clarify (and also the `riverpod_generator` dependency).
> > > Retrieving the serialised authData from the secure storage is what I did in Bstim project(not needed for this PR), riverpod is unnecessary, I've removed it.
> 
> I guess the logic for the popup login would be something like:
> 
> 1. try to retrieve the `authData` from secure storage
> 2. if step 1 is successful, do nothing
> 3. otherwise, show the popup login window and do the normal authentication (this is what you have implemented)
> 
> Is this the agreed logic @junhaow1 and @gjwgit ?
> 
> We may need to check token expiration and refresh the access token if expired. But this could be for a different issue.

Yes, implementing now. We could merge this PR and I will work on another branch

**Pull request merged**